### ISSUE TITLE
[Issue] Stuck at log screen

### ISSUE THREAD

**Issue by shakeabi** [end-4/dots-hyprland#312]
- **(For usage) I have read the [wiki](https://end-4.github.io/dots-hyprland-wiki/en/i-i/02usage)**
- **(For troubleshooting) I have read the [wiki](https://end-4.github.io/dots-hyprland-wiki/en/i-i/03troubleshooting)**
- Linux distro: EndeavourOS

---
## The issue
I'm pretty new to hyprland. I had installed dots-hyprland using the script.
However, after installation when I logout, choose hyprland and try to login, it keeps going back to the log screen. I've attached the crash logs below.

More Details:
I had installed hyprland with i3. 
Upon googling many suggested that I had to remove kde plasma which comes default with endeavour which I also did.
The error logs seems to not find my GPU? I have to mention that my laptop does not have a dedicated GPU, just the intel integrated one.

Any thoughts on how to get this to work? TIA!

<details>
<summary>Click to expand</summary>

```plain
--------------------------------------------
   Hyprland Crash Report
--------------------------------------------
All these computers...

Hyprland received signal 6 (Aborted)

Version: e52d3fa8522ff83813a9006d4bf46b3057f4a4cc
Tag: v0.36.0-71-ge52d3fa8

System info:
	System name: Linux
	Node name: abishake-latitude5410-eos
	Release: 6.7.8-arch1-1
	Version: #1 SMP PREEMPT_DYNAMIC Sun, 03 Mar 2024 00:30:36 +0000

GPU:
	00:02.0 VGA compatible controller [0300]: Intel Corporation CometLake-U GT2 [UHD Graphics] [8086:9b41] (rev 02) (prog-if 00 [VGA controller])


os-release:
	NAME="EndeavourOS"
	PRETTY_NAME="EndeavourOS"
	ID="endeavouros"
	ID_LIKE="arch"
	BUILD_ID=rolling
	ANSI_COLOR="38;2;23;147;209"
	HOME_URL="https://endeavouros.com"
	DOCUMENTATION_URL="https://discovery.endeavouros.com"
	SUPPORT_URL="https://forum.endeavouros.com"
	BUG_REPORT_URL="https://forum.endeavouros.com/c/arch-based-related-questions/bug-reports"
	PRIVACY_POLICY_URL="https://endeavouros.com/privacy-policy-2"
	LOGO="endeavouros"
	


Backtrace:
	#0 | Hyprland(_Z12getBacktracev+0x62) [0x63b827825f32]
		getBacktrace()
		??:?
	#1 | Hyprland(_ZN13CrashReporter18createAndSaveCrashEi+0x6d3) [0x63b8277e6933]
		CrashReporter::createAndSaveCrash(int)
		/usr/src/debug/hyprland-git/Hyprland/build/../src/debug/CrashReporter.cpp:106
	#2 | Hyprland(_Z25handleUnrecoverableSignali+0x48) [0x63b82777d538]
		handleUnrecoverableSignal(int)
		/usr/src/debug/hyprland-git/Hyprland/build/../src/Compositor.cpp:35
	#3 | /usr/lib/libc.so.6(+0x3c770) [0x7bd08b76e770]
		??
		??:0
	#4 | /usr/lib/libc.so.6(+0x8d32c) [0x7bd08b7bf32c]
		__static_initialization_and_destruction_0() [clone .lto_priv.19]
		/usr/src/debug/hyprland-git/Hyprland/build/../src/managers/ProtocolManager.hpp:26
	#5 | /usr/lib/libc.so.6(gsignal+0x18) [0x7bd08b76e6c8]
		??
		??:0
	#6 | /usr/lib/libc.so.6(abort+0xd7) [0x7bd08b7564b8]
		??
		??:0
	#7 | /usr/lib/libstdc++.so.6(+0x9ca6f) [0x7bd08ba9ca6f]
		__static_initialization_and_destruction_0() [clone .lto_priv.43]
		/usr/src/debug/hyprland-git/Hyprland/build/../src/helpers/Watchdog.hpp:32 (discriminator 1)
	#8 | /usr/lib/libstdc++.so.6(+0xb011c) [0x7bd08bab011c]
		void Debug::log<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int const&>(LogLevel, std::basic_format_string<char, std::type_identity<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&>::type, std::type_identity<int const&>::type>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int const&)
		/usr/src/debug/hyprland-git/Hyprland/build/../src/debug/Log.hpp:72
	#9 | /usr/lib/libstdc++.so.6(+0xb0189) [0x7bd08bab0189]
		void Debug::log<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int const&>(LogLevel, std::basic_format_string<char, std::type_identity<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&>::type, std::type_identity<int const&>::type>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int const&)
		/usr/src/debug/hyprland-git/Hyprland/build/../src/debug/Log.hpp:76 (discriminator 1)
	#10 | /usr/lib/libstdc++.so.6(+0xb03ed) [0x7bd08bab03ed]
		CCompositor::startCompositor()
		/usr/src/debug/hyprland-git/Hyprland/build/../src/Compositor.cpp:615
	#11 | Hyprland(+0x6d54b) [0x63b82773e54b]
		throwError(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
		/usr/src/debug/hyprland-git/Hyprland/build/../src/helpers/MiscFunctions.cpp:799 (discriminator 1)
	#12 | Hyprland(_ZN11CCompositor10initServerEv+0x697) [0x63b827784de7]
		CCompositor::initServer()
		/usr/src/debug/hyprland-git/Hyprland/build/../src/Compositor.cpp:126 (discriminator 2)
	#13 | Hyprland(main+0x79e) [0x63b82776585e]
		main
		/usr/src/debug/hyprland-git/Hyprland/build/../src/main.cpp:105 (discriminator 1)
	#14 | /usr/lib/libc.so.6(+0x25cd0) [0x7bd08b757cd0]
		??
		??:0
	#15 | /usr/lib/libc.so.6(__libc_start_main+0x8a) [0x7bd08b757d8a]
		??
		??:0
	#16 | Hyprland(_start+0x25) [0x63b82777a4c5]
		_start
		??:?


Log tail:
[LOG] Hyprland PID: 20695
[LOG] ===== SYSTEM INFO: =====
[LOG] System name: Linux
[LOG] Node name: abishake-latitude5410-eos
[LOG] Release: 6.7.8-arch1-1
[LOG] Version: #1 SMP PREEMPT_DYNAMIC Sun, 03 Mar 2024 00:30:36 +0000


[LOG] GPU information:
00:02.0 VGA compatible controller [0300]: Intel Corporation CometLake-U GT2 [UHD Graphics] [8086:9b41] (rev 02) (prog-if 00 [VGA controller])


[LOG] os-release:
NAME="EndeavourOS"
PRETTY_NAME="EndeavourOS"
ID="endeavouros"
ID_LIKE="arch"
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://endeavouros.com"
DOCUMENTATION_URL="https://discovery.endeavouros.com"
SUPPORT_URL="https://forum.endeavouros.com"
BUG_REPORT_URL="https://forum.endeavouros.com/c/arch-based-related-questions/bug-reports"
PRIVACY_POLICY_URL="https://endeavouros.com/privacy-policy-2"
LOGO="endeavouros"

[LOG] ========================



[INFO] If you are crashing, or encounter any bugs, please consult https://wiki.hyprland.org/Crashes-and-Bugs/


[LOG] 
Current splash: "I use Arch, btw" - John Cena


[LOG] Creating the HookSystem!
[LOG] Creating the KeybindManager!
[LOG] [hookSystem] New hook event registered: configReloaded
[LOG] Creating the AnimationManager!
[LOG] Created a bezier curve, baked 255 points, mem usage: 4.08kB, time to bake: 62.20µs. Estimated average calc time: 0.10µs.
[LOG] Creating the ConfigManager!
[LOG] NOTE: further logs to stdout / logfile are disabled by default. Use debug:disable_logs and debug:enable_stdout_logs to override this.
[LOG] Created a bezier curve, baked 255 points, mem usage: 4.08kB, time to bake: 39.23µs. Estimated average calc time: 0.09µs.
[LOG] Using config: /home/abishake/.config/hypr/hyprland.conf
[LOG] Creating the CHyprError!
[LOG] [hookSystem] New hook event registered: focusedMon
[LOG] [hookSystem] New hook event registered: preRender
[LOG] Creating the LayoutManager!
[LOG] [hookSystem] New hook event registered: preConfigReload
[LOG] Created a bezier curve, baked 255 points, mem usage: 4.08kB, time to bake: 38.65µs. Estimated average calc time: 0.08µs.
[LOG] Using config: /home/abishake/.config/hypr/hyprland.conf
[LOG] Created a bezier curve, baked 255 points, mem usage: 4.08kB, time to bake: 39.81µs. Estimated average calc time: 0.09µs.
[LOG] Created a bezier curve, baked 255 points, mem usage: 4.08kB, time to bake: 38.91µs. Estimated average calc time: 0.09µs.
[LOG] Created a bezier curve, baked 255 points, mem usage: 4.08kB, time to bake: 38.73µs. Estimated average calc time: 0.08µs.
[LOG] Created a bezier curve, baked 255 points, mem usage: 4.08kB, time to bake: 38.82µs. Estimated average calc time: 0.09µs.
[LOG] Created a bezier curve, baked 255 points, mem usage: 4.08kB, time to bake: 38.71µs. Estimated average calc time: 0.09µs.
[LOG] Created a bezier curve, baked 255 points, mem usage: 4.08kB, time to bake: 38.77µs. Estimated average calc time: 0.09µs.
[LOG] Created a bezier curve, baked 255 points, mem usage: 4.08kB, time to bake: 38.74µs. Estimated average calc time: 0.09µs.
[LOG] Created a bezier curve, baked 255 points, mem usage: 4.08kB, time to bake: 38.64µs. Estimated average calc time: 0.09µs.
[LOG] Created a bezier curve, baked 255 points, mem usage: 4.08kB, time to bake: 39.26µs. Estimated average calc time: 0.08µs.
[LOG] Created a bezier curve, baked 255 points, mem usage: 4.08kB, time to bake: 38.60µs. Estimated average calc time: 0.09µs.
[LOG] Created a bezier curve, baked 255 points, mem usage: 4.08kB, time to bake: 38.78µs. Estimated average calc time: 0.08µs.
[LOG] Disabling stdout logs! Check the log for further logs.
[backend/backend.c:104] Timeout waiting session to become active
[backend/backend.c:406] Failed to start a DRM session
[CRITICAL] m_sWLRBackend was NULL! This usually means wlroots could not find a GPU or enountered some issues.
[CRITICAL] Critical error thrown: wlr_backend_autocreate() failed!

```

</details>

**Comment by clsty** [comment]
The install script don't handle any GPU thing. You need to solve it by yourself.

**Comment by shakeabi** [comment]
Figured out the issue. The issue was hyprland doesn't work well with lightdm. Since endevour+i3 comes with i3, it kept crashing. Replacing window manager with gdm fixed the issue.

tldr; Newbies, please follow the wiki as it is by installing endeavour+**GNOME** (Gnome comes with gdm and not lightdm)

**Comment by clsty** [comment]
I'm glad you have figured it out by yourself.

> tldr; Newbies, please follow the wiki as it is by installing endeavour+**GNOME** (Gnome comes with gdm and not lightdm)

It's not necessary to use a DM though. You can also use a tty or greetd. (e.g. I use tty and end-4 use greetd)
P.S. end-4 now uses CachyOS which is another Arch-based distro, and I still use vanilla Arch. Both work well.
